account_invoice_default_account_date/models/account_invoice.py

```diff
@@ -0,0 +1,20 @@
+from datetime import date
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def write(self, vals):
+        # vendor bills
+        if (
+            not self.date
+            and not vals.get("date")
+            and self.type in ("in_invoice", "in_refund")
+        ):
+
+            vals["date"] = date.today()
```

Comment:
Prefer using

```py
from odoo.models import Date
vals["date"] = Date.context_today()
```

It will localize the date. It could matter if the timezone difference sends you across midnight.

Reply:
changed with : 
```python
from odoo import fields
vals["date"] = fields.Date.context_today(self)
```